Dart/src/com/jetbrains/lang/dart/ide/runner/server/DartCommandLineRunningState.java

```diff
@@ -197,10 +209,11 @@ private void setupParameters(@NotNull final Project project,
     if (customObservatoryPort > 0) {
       myObservatoryPort = customObservatoryPort;
     }
-    else {
-      myObservatoryPort = PubServerManager.findOneMoreAvailablePort(myDebuggingPort);
-      commandLine.addParameter("--enable-vm-service:" + myObservatoryPort);
-    }
+    // TODO Enable this for debugging package:test code, but make sure it is for the test runner VM.
+    //else {
+    //  myObservatoryPort = PubServerManager.findOneMoreAvailablePort(myDebuggingPort);
+    //  commandLine.addParameter("--enable-vm-service:" + myObservatoryPort);
+    //}
```

Comment:
This disables observatory for standard Dart command line app as well. I've added `instanceof DartTestRunnerParameters` check instead.